src/System.Net.Http/src/System/Net/Http/Unix/CurlHandler.cs

```diff
@@ -202,6 +205,23 @@ internal DecompressionMethods AutomaticDecompression
             }
         }
 
+        internal bool PreAuthenticate
+        {
+            get
+            {
+                return _preAuthenticate;
+            }
+            set
+            {
+                CheckDisposedOrStarted();
+                _preAuthenticate = value;
+                if (value)
+                {
+                    _credentialCache = new CredentialCache();
+                }
+            }
+        }
```

Comment:
I know that HttpClient is meant to be used concurrently, and you're not allowed to reconfigure it after any operation has been started (hence the CheckDisposedOrStarted checks).  Of course, while one thread is setting PreAuthenticate, another could be starting an operation, and there's a race condition as to whether that'll throw or not based on whether the first thread setting PreAuthenticate is past the CheckDisposedOrStarted call.  Do we care about such race conditions, or is it just expected / documented that all property setting must be done prior to any attempts to start an operation?